xla/service/gpu/transforms/dynamic_slice_fusion_rewriter_test.cc

```diff
@@ -2143,4 +2143,53 @@ TEST_F(DynamicSliceFusionRewriterTest, ReduceScatterDynamicSlice) {
   RunAndFilecheckHloRewrite(hlo, DynamicSliceFusionRewriter("gpu"), expected);
 }
 
+TEST_F(DynamicSliceFusionRewriterTest, OffsetAsFunctionOfIndvar) {
+  const char* hlo = R"(
+    HloModule test, replica_count=2
```

Comment:
Seems like a lot of HLO---do we need all of it to exhibit the behaviour you intend to test?

Reply:
Unfortunately, yes. I need the while loop, the collective, and a simple (yet random) seq of operations for making sure it is being fused as a general pattern.